### PR TITLE
Feature - Handling empty anchors

### PR DESCRIPTION
**Pull request by dav009**
There are links with empty descriptions.
Looking at the code there is a comment When anchor is empty, it means is the same of id.
Which sort of makes sense since they look like: [[wikipedia_article]]. [1]

getAnchor()  returned the right anchor. However when the object got serialised to json the description field will be empty ( since getAnchor() would not be called)

-----
**Edit**

Pom Changes:
We were getting some Links without wikiIds. One of the reason was that jwpl was incorrectly handling `gallery` markup. So I fixed jwpl and updated the dependency on `jsonpedia`

There is still another case which could return empty WikiIds. But there are enough changes on this PR already.
 
[1] https://github.com/diegoceccarelli/json-wikipedia/blob/master/src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java#L53

### CODE REVIEW

File: src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java [hunk 1]
```diff
@@ -38,9 +38,9 @@
 	public Link(String id, String anchor, int start, int end) {
 		super();
 		this.id = id;
-		this.anchor = anchor;
 		this.start = start;
 		this.end = end;
+        setAnchor(anchor);
```
Comment (keynmol) [inline src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java]: indentation

Comment (dav009) [inline src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java]: this is weird on intelliJ it shows with the same number of spaces..

Comment (dav009) [inline src/main/java/it/cnr/isti/hpc/wikipedia/article/Link.java]: screen shot 2015-06-23 at 10 43 08

**Comment by dav009** [comment]
@keynmol  @Lugrin re re? 
please try cloning and doing: `mvn assembly:assembly` I want to make sure this compiles somewhere outside my machine.

**Comment by keynmol** [comment]
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 8.178 s
[INFO] Finished at: 2015-06-23T14:17:34+01:00
[INFO] Final Memory: 7M/123M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal on project json-wikipedia: Could not resolve dependencies for project it.cnr.isti.hpc:json-wikipedia:jar:1.0.0: The following artifacts could not be resolved: de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.datamachine:jar:1.1.0-SNAPSHOT, de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.revisionmachine:jar:1.1.0-SNAPSHOT, de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.timemachine:jar:1.1.0-SNAPSHOT, de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.parser:jar:1.1.0-SNAPSHOT: Could not find artifact de.tudarmstadt.ukp.wikipedia:de.tudarmstadt.ukp.wikipedia.datamachine:jar:1.1.0-SNAPSHOT in local jars (file:///Users/velvetbaldmime/idio/json-wikipedia/lib) -> [Help 1]
```

**Pull request merged**